### ISSUE TITLE
Considering an improvement with mamba2?

### ISSUE THREAD

**Issue by zzzendurance** [state-spaces/mamba#435]
Hello author, I have seen that your BiMamba code is an improvement on the basis of Mamba1 (similar to Vision Mamba), Mamba2 has been released recently, may I ask if you have considered using Mamba2 to improve it? Can you share the idea of BiMamba2?

**Comment by zzzendurance** [comment]
![Uploading image.png…]()
The first and third layers represent the number of two-way mamba layers I used, and m1 and m2 represent mamba1 and mamba2, respectively.

After I used vision mamba (doing bi-directional processing internally), the number of parameters obviously decreased a lot.

Of course, I know that this is my own work to do (doing two-way processing within mamba2), but I still want to ask why this situation has occurred, and whether there are other similar open source work now.

**Comment by albertfgu** [comment]
Yes, our official bidirectional version will be released next week, stay tuned!

**Comment by jeohalves** [comment]
Is the official bidirectional version already available? If so, how can I use it? Thanks!

**Comment by albertfgu** [comment]
yes, it's here: https://github.com/goombalab/hydra
